### PR TITLE
plan: resolve show stmt in plan builder.

### PR DESCRIPTION
**Pull request by hanfei1991**
This is one step of removing resolver.
@shenli @coocood @zimulala PTAL

### CODE REVIEW

File: plan/planbuilder.go [hunk 1]
```diff
@@ -556,3 +557,89 @@ func buildShowEventsSchema() expression.Schema {
 	schema = append(schema, buildColumn(tblName, "Database Collation", mysql.TypeVarchar, 32))
 	return schema
 }
+
+func getShowColNamesAndTypes(s *ast.ShowStmt) (names []string, ftypes []byte) {
```
Comment (coocood) [inline plan/planbuilder.go]: Add comment about return empty ftypes.

File: plan/planbuilder.go [hunk 2]
```diff
@@ -556,3 +557,90 @@ func buildShowEventsSchema() expression.Schema {
...
+func buildShowDefaultSchema(s *ast.ShowStmt) expression.Schema {
+	names, ftypes := getShowColNamesAndTypes(s)
+	schema := make(expression.Schema, 0, len(names))
+	for i, name := range names {
+		col := &expression.Column{
+			ColName: model.NewCIStr(name),
+		}
+		var retType types.FieldType
+		if len(ftypes) == 0 || ftypes[i] == 0 {
+			// use varchar as the default return column type
```
Comment (zimulala) [inline plan/planbuilder.go]: Complete this comment.

File: plan/planbuilder.go [hunk 3]
```diff
@@ -556,3 +557,90 @@ func buildShowEventsSchema() expression.Schema {
...
+		names = table.ColDescFieldNames(s.Full)
+	case ast.ShowWarnings:
+		names = []string{"Level", "Code", "Message"}
+		ftypes = []byte{mysql.TypeVarchar, mysql.TypeLong, mysql.TypeVarchar}
+	case ast.ShowCharset:
+		names = []string{"Charset", "Description", "Default collation", "Maxlen"}
+		ftypes = []byte{mysql.TypeVarchar, mysql.TypeVarchar, mysql.TypeVarchar, mysql.TypeLonglong}
+	case ast.ShowVariables:
+		names = []string{"Variable_name", "Value"}
+	case ast.ShowStatus:
```
Comment (zimulala) [inline plan/planbuilder.go]: Use `case ast.ShowVariables, ast.ShowStatus:` directly.

**Comment by hanfei1991** [comment]
@zimulala PTAL

File: plan/planbuilder.go [hunk 4]
```diff
@@ -556,3 +557,88 @@ func buildShowEventsSchema() expression.Schema {
...
+	case ast.ShowCreateDatabase:
+		names = []string{"Database", "Create Database"}
+	case ast.ShowGrants:
+		names = []string{fmt.Sprintf("Grants for %s", s.User)}
+	case ast.ShowTriggers:
+		names = []string{"Trigger", "Event", "Table", "Statement", "Timing", "Created",
+			"sql_mode", "Definer", "character_set_client", "collation_connection", "Database Collation"}
+		ftypes = []byte{mysql.TypeVarchar, mysql.TypeVarchar, mysql.TypeVarchar, mysql.TypeVarchar, mysql.TypeVarchar, mysql.TypeVarchar,
+			mysql.TypeVarchar, mysql.TypeVarchar, mysql.TypeVarchar, mysql.TypeVarchar, mysql.TypeVarchar}
+	case ast.ShowProcedureStatus, ast.ShowEvents:
```
Comment (zimulala) [inline plan/planbuilder.go]: The `ShowTriggers`, `ShowEvents` and `ast.ShowProcedureStatus` are handle by line 390-395.
Can we remove one of them？

**Comment by zimulala** [comment]
Does it need tests?

**Comment by hanfei1991** [comment]
@zimulala Old test can cover this :)

**Comment by zimulala** [comment]
Old test only cover some of these. @hanfei1991

**Comment by zimulala** [comment]
LGTM

**Comment by hanfei1991** [comment]
@coocood PTAL

**Comment by coocood** [comment]
LGTM

**Pull request merged**